diff --git a/runtime/bin/gen_snapshot.cc b/runtime/bin/gen_snapshot.cc
index ab04917d..4e41b496 100644
--- a/runtime/bin/gen_snapshot.cc
+++ b/runtime/bin/gen_snapshot.cc
@@ -969,5 +969,12 @@ int main(int argc, char** argv) {
 }  // namespace dart
 
 int main(int argc, char** argv) {
-  return dart::bin::main(argc, argv);
+  const bool argv_converted = dart::bin::ShellUtils::GetUtf8Argv(argc, argv);
+  const int result = dart::bin::main(argc, argv);
+  if (argv_converted) {
+    for (int i = 0; i < argc; i++) {
+      free(argv[i]);
+    }
+  }
+  return result;
 }
